### PR TITLE
Prevent tabbing out of media modal

### PR DESCRIPTION
**Pull request by KTS915**
This PR prevents tabbing out of the modal that opens when editing an uploaded media file. This is important for reasons of accessibility.

## Description
Previous PRs for CP v2.1 replaced the jQuery `tabbables` object with a functionally-equivalent array in vanilla JavaScript. But one of those objects was missed. This PR rectifies that oversight.

## Motivation and context
If you go the media library, set it in a grid formation, and then select an item to edit, a modal will open. Those who cannot use a mouse will use the Tab key to navigate through that modal, and it is important that they cannot "tab out" of the modal to somewhere else on the screen. (In other words, no matter how many times they hit the Tab key, they should always remain within the modal.)

Previous PRs have made it possible for tabbing out to occur. This PR fixes that.

## How has this been tested?
On my localhost test site.

## Types of changes
- Bug fix

### CODE REVIEW

File: src/wp-includes/js/media-views.js [hunk 1]
```diff
@@ -4409,13 +4409,20 @@
...
-		 * @return {Object} A jQuery collection of tabbable elements.
+		 * @return {Array} A JavaScript array of tabbable elements.
 		 */
 		getTabbables: function() {
+			var tabbables = [ ...document.querySelector( '.media-modal' ).querySelectorAll( 'a[href], button, input, textarea, select, [tabindex]' ) ];
+
 			// Skip the file input added by Plupload.
-			return this.$( ':tabbable' ).not( '.moxie-shim input[type="file"]' );
+			for ( var i = 0, n = tabbables.length; i < n; i++ ) {
+				if ( tabbables[i].tagName === 'input' && tabbables[i].className.includes( 'moxie-shim' ) && tababbles[i].type === 'file' ) {
```
Comment (mattyrob) [inline src/wp-includes/js/media-views.js:4421]: Typo here in the last comparison - `tababbles[i].type === 'file'` noted that 'tababbles' is misspelled.

**Review by mattyrob — APPROVED** [review]

File: src/wp-includes/js/media-views.js (hunk 1 above)
Comment (mattyrob) [reply src/wp-includes/js/media-views.js]: I have pushed a change that should resolve this.

Comment (KTS915) [reply src/wp-includes/js/media-views.js]: Thanks for spotting and resolving that!

**Comment by KTS915** [comment]
This requires a bit more work so that it addresses a similar problem with some widgets, so please don't merge yet.

**Comment by mattyrob** [comment]
@KTS915 - any thoughts or feedback on my most recent commit?

https://github.com/ClassicPress/ClassicPress/pull/1363/commits/281484d39416fb85a6f189b8349a6ca6564f625b

`src/wp-includes/js/media-views.js` is built from numerous other files and the areas to change in the pre-build files seems to be isolated to `src/wp-includes/js/media/views/focus-manager.js` - I've updated there to make sure `grunt rollup` still builds in the same way and moved the `isVisible()` function into score and added some documentation.

**Comment by KTS915** [comment]
@mattyrob I had no idea that `src/wp-includes/js/media-views.js` is built from other files. What identifies such a file? (It would obviously be better if I made changes to the originating file in future.)

So far as the changes themselves go, I see you've added a `this` in a couple of spots. This is working well for widgets, but tabbing out is once again possible when editing a media item from the media grid view. That's because, on that screen, the code is producing the following message: `Uncaught TypeError: this.isVisible is not a function`.

I'll investigate further.

**Comment by KTS915** [comment]
Ah, I think I've worked it out! It's the usual problem with the JavaScript `this`! I'll add a commit to fix it!

**Comment by mattyrob** [comment]
As discussed, merge for wider testing.

**Pull request merged**